### ISSUE TITLE
Possible regression in newer Mesa builds?

### ISSUE THREAD

**Issue by lorn10** [iXit/wine-nine-standalone#111]
### System information
- OS:
```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel version:
```
Linux test-iMac 5.4.0-73-generic
```
- GPU:
```
  *-display                 
       description: VGA compatible controller
       product: Whistler [Radeon HD 6730M/6770M/7690M XT]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:01:00.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom
       configuration: driver=radeon latency=0
       resources: irq:50 memory:90000000-9fffffff memory:a8800000-a881ffff ioport:2000(size=256) memory:a8820000-a883ffff
```

Hi all!

I have most likely found a wrong behavior of Gallium Nine in newer Mesa builds. For example, the original Xbox game **Baldur's Gate Dark Alliance** seems wrongly rendered. This is definitely not an issue in the emulator Cxbx-Reloaded. I have tested different Cxbx-R builds and the problem always occurs only in conjunction with more recent Mesa versions. So this seems to be effectively a problem of newer Gallium Nine releases.

**Mesa 20.2.6**, the background is correctly rendered:
![Screenshot_Mesa_20 2 6](https://user-images.githubusercontent.com/9931771/120188406-6127cf00-c216-11eb-89a5-a845e4e72f84.png)

**Mesa 21.2 devel**, the background is missing:
![Screenshot_Mesa_21 2_devel](https://user-images.githubusercontent.com/9931771/120188526-861c4200-c216-11eb-9513-ada5b09868f0.png)

Regarding this game, also the intro scenes, which are not 3D based, are missing. However, when I enter the game, everything which is 3D seems to be correctly handled. Other games are running normally also at newer Mesa builds.

Are there any newer kernel requirements for the new 32bit DX9 stuff in Gallium Nine? As mentioned above, I tested this on  5.4.0-73 but I am now on 5.8.0-53. ;-)

I know this is a somewhat difficult topic, you have first to get the Cxbx-R emulator and install it in Wine. Then you have to find the Xbox game in the Web and prepare it to get it compatible with the emulator. All in all that is not to be underestimated ... Whatever, I am ready to help, just ask.... 

The Wine part is easy, I installed the latest development version, every release since 6.8 should work.

~~But you will need for the mentioned game, a special HLSL "pixel shader build" of Cxbx-R, you can get that one here (GitHub login needed): https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/suites/2640806238/artifacts/57974673~~

**Update 15.07.2021:**
The HLSL "pixel shader" feature was in the meantime merged into the regular release:
https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/releases

Then start it from a new, clean wine prefix which contains the `d3dcompiler_47` and `galliumnine` options.

```
WINEARCH=win64 WINEPREFIX=~/.cxbx winecfg

export WINEPREFIX="/home/<youruserprofile>/.cxbx/"

winetricks -q d3dcompiler_47

winetricks -q galliumnine
```
To launch the emulator from the CLI I use the following command:

`WINEPREFIX=~/.cxbx wine /home/<youruserprofile>/Downloads/Cxbx/cxbx.exe`

**Comment by axeldavy** [comment]
We fixed a number of issues between the two mesa releases, but possibly there has been some regressions.
Ideally you'd do a bisection on the mesa tree to find the regressing commit.
A trace can be useful (see wiki) to find a fix that helps with both what was fixed by the regressing commit and that game.
With a trace I can do a bisection myself (but I'm pretty busy and won't be able to work on it soon).

To ease the bisection, the trace must be small and replay fast (minimum level of detail to reproduce the issue, capped fps and resolution, etc).

**Comment by lorn10** [comment]
Okay thanks @axeldavy for this information. I will try first out the very latest oibaf Mesa drivers and report back the result, maybe there is meanwhile already a change in that topic. And many thanks for your GREAT work in that Gallium Nine project. :+1:

**Comment by lorn10** [comment]
@CTRL-ALT-DEL-92 
Thanks for your feedback! :+1:  

Just to be clear, - I am not 100% sure if this issue here is really only 2D related, Well, this could be. But the background of the here described Xbox game **Baldur's Gate Dark Alliance** is definitely somewhat "animated" and it looks more then just 2D. Whatever, the also missing intro FMV sequences are definitely 2D based. So the underlying problem could be really somewhere at the 2D rendering.

**Comment by CTRL-ALT-DEL-92** [comment]
Just to be clear this is my situation with oblivion:
![oblivion](https://user-images.githubusercontent.com/78966095/125680855-a9293760-a781-40fc-89fb-5074cd914340.png)

the intro video is not working as well.  The 3D rendering is working fine

**Comment by lorn10** [comment]
This really looks similar to my issue. Are you able to apply a bisection? https://wiki.ubuntu.com/X/Bisecting

The devs may need this to identify the effective root of the problem. Alternatively, they must organize the affected game and do this bisection-stuff at their own.

Well, I am still to much noob to do a bisection, sorry. But I have "prepared" everything to help, if a g9 def is interested he should give me a PM. I will then replay back with all necessary links to get the "needed stuff" and resolve this problem. ;-)

**Comment by axeldavy** [comment]
Could you try with software rendering if the bug disappears ? For that you need to launch with the command line and the env var D3D_ALWAYS_SOFTWARE=1

To debug, we'd need to reproduce the issue, for that you could make a short trace (https://github.com/iXit/wine-nine-standalone/wiki/apitrace).

**Comment by CTRL-ALT-DEL-92** [comment]
Dear G9 devs,

I tried to run Oblivion with the env var D3D_ALWAYS_SOFTWARE=1 but nothing changed. I wasn't able to produce an apitrace though... Here is the log file from crossover I don't know if can be of any use for you: Running command: "C:\\users\\crossover\\Start Menu\\Oblivion.lnk"
Bottle: nine
Debug channels: 
Extra environment variables: D3D_ALWAYS_SOFTWARE=1

CrossOver 21.0.0~beta1
Install location: /opt/cxoffice
64-bit OS
Desktop session: xfce
Window manager: Xfwm4
Locale: it_IT.UTF-8

Contents of /etc/os-release:
NAME="Linux Mint"
VERSION="20.2 (Uma)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 20.2"
VERSION_ID="20.2"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=uma
UBUNTU_CODENAME=focal

cxdiag output:
[Properties]
"vulkan.present"="yes"

[MissingGStreamer1Bad2]
"Title"="The gst-plugins-bad 32-bit GStreamer plugins appear to be missing h264parse"

[MissingGStreamer1Libav]
"Title"="The gst-libav 32-bit GStreamer plugins appear to be missing avdec_eac3"

[Properties]
"display.depth"="24"

[Properties]
"display.dpi"="-1.0"

[Properties]
"opengl.vendor"="Intel"

[Properties]
"opengl.version"="2.1 Mesa 21.3.0-devel (git-cb589ee 2021-07-16 focal-oibaf-ppa)"

[Properties]
"opengl.renderer"="Mesa Intel(R) HD Graphics (ILK)"

[Properties]
"opengl.glsl_version"="1.20"

[Properties]
"opengl.extensions"="GL_ARB_multisample GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_copy_texture GL_EXT_subtexture GL_EXT_texture_object GL_EXT_vertex_array GL_EXT_compiled_vertex_array GL_EXT_texture GL_EXT_texture3D GL_IBM_rasterpos_clip GL_ARB_point_parameters GL_EXT_draw_range_elements GL_EXT_packed_pixels GL_EXT_point_parameters GL_EXT_rescale_normal GL_EXT_separate_specular_color GL_EXT_texture_edge_clamp GL_SGIS_generate_mipmap GL_SGIS_texture_border_clamp GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_ARB_framebuffer_sRGB GL_ARB_multitexture GL_EXT_framebuffer_sRGB GL_IBM_multimode_draw_arrays GL_IBM_texture_mirrored_repeat GL_3DFX_texture_compression_FXT1 GL_ARB_texture_cube_map GL_ARB_texture_env_add GL_ARB_transpose_matrix GL_EXT_blend_func_separate GL_EXT_fog_coord GL_EXT_multi_draw_arrays GL_EXT_secondary_color GL_EXT_texture_env_add GL_EXT_texture_filter_anisotropic GL_EXT_texture_lod_bias GL_INGR_blend_func_separate GL_NV_blend_square GL_NV_light_max_exponent GL_NV_texgen_reflection GL_NV_texture_env_combine4 GL_S3_s3tc GL_SUN_multi_draw_arrays GL_ARB_texture_border_clamp GL_ARB_texture_compression GL_EXT_framebuffer_object GL_EXT_texture_compression_s3tc GL_EXT_texture_env_combine GL_EXT_texture_env_dot3 GL_MESA_window_pos GL_NV_packed_depth_stencil GL_NV_texture_rectangle GL_ARB_depth_texture GL_ARB_occlusion_query GL_ARB_shadow GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_mirrored_repeat GL_ARB_window_pos GL_ATI_fragment_shader GL_EXT_stencil_two_side GL_EXT_texture_cube_map GL_NV_depth_clamp GL_NV_fog_distance GL_NV_half_float GL_APPLE_packed_pixels GL_ARB_draw_buffers GL_ARB_fragment_program GL_ARB_fragment_shader GL_ARB_shader_objects GL_ARB_vertex_program GL_ARB_vertex_shader GL_ATI_draw_buffers GL_ATI_texture_env_combine3 GL_ATI_texture_float GL_EXT_shadow_funcs GL_EXT_stencil_wrap GL_MESA_pack_invert GL_NV_primitive_restart GL_ARB_depth_clamp GL_ARB_fragment_program_shadow GL_ARB_half_float_pixel GL_ARB_occlusion_query2 GL_ARB_point_sprite GL_ARB_shading_language_100 GL_ARB_sync GL_ARB_texture_non_power_of_two GL_ARB_vertex_buffer_object GL_ATI_blend_equation_separate GL_EXT_blend_equation_separate GL_OES_read_format GL_ARB_color_buffer_float GL_ARB_pixel_buffer_object GL_ARB_texture_compression_rgtc GL_ARB_texture_float GL_ARB_texture_rectangle GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_shared_exponent GL_ARB_framebuffer_object GL_EXT_framebuffer_blit GL_EXT_packed_depth_stencil GL_ARB_vertex_array_object GL_ATI_separate_stencil GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_gpu_program_parameters GL_EXT_gpu_shader4 GL_EXT_texture_array GL_EXT_texture_integer GL_EXT_texture_sRGB_decode GL_EXT_timer_query GL_OES_EGL_image GL_EXT_texture_buffer_object GL_ARB_copy_buffer GL_ARB_depth_buffer_float GL_ARB_draw_instanced GL_ARB_half_float_vertex GL_ARB_instanced_arrays GL_ARB_map_buffer_range GL_ARB_texture_rg GL_ARB_texture_swizzle GL_ARB_vertex_array_bgra GL_EXT_texture_swizzle GL_EXT_vertex_array_bgra GL_NV_conditional_render GL_AMD_conservative_depth GL_AMD_seamless_cubemap_per_texture GL_ARB_ES2_compatibility GL_ARB_blend_func_extended GL_ARB_debug_output GL_ARB_draw_elements_base_vertex GL_ARB_explicit_attrib_location GL_ARB_fragment_coord_conventions GL_ARB_provoking_vertex GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_shader_texture_lod GL_ARB_texture_query_lod GL_ARB_texture_rgb10_a2ui GL_ARB_vertex_type_2_10_10_10_rev GL_EXT_provoking_vertex GL_EXT_texture_snorm GL_MESA_texture_signed_rgba GL_NV_copy_image GL_NV_texture_barrier GL_ARB_get_program_binary GL_ARB_robustness GL_ARB_separate_shader_objects GL_ARB_shader_bit_encoding GL_EXT_direct_state_access GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_base_instance GL_ARB_compressed_texture_pixel_storage GL_ARB_conservative_depth GL_ARB_internalformat_query GL_ARB_map_buffer_alignment GL_ARB_shading_language_420pack GL_ARB_shading_language_packing GL_ARB_texture_storage GL_AMD_shader_trinary_minmax GL_ARB_arrays_of_arrays GL_ARB_clear_buffer_object GL_ARB_copy_image GL_ARB_explicit_uniform_location GL_ARB_invalidate_subdata GL_ARB_program_interface_query GL_ARB_texture_query_levels GL_ARB_vertex_attrib_binding GL_KHR_debug GL_KHR_texture_compression_astc_ldr GL_ARB_buffer_storage GL_ARB_clear_texture GL_ARB_enhanced_layouts GL_ARB_internalformat_query2 GL_ARB_multi_bind GL_ARB_seamless_cubemap_per_texture GL_ARB_shader_group_vote GL_ARB_shading_language_include GL_ARB_texture_mirror_clamp_to_edge GL_ARB_vertex_type_10f_11f_11f_rev GL_EXT_shader_integer_mix GL_ARB_clip_control GL_ARB_get_texture_sub_image GL_ARB_texture_barrier GL_EXT_polygon_offset_clamp GL_KHR_blend_equation_advanced GL_KHR_context_flush_control GL_ARB_parallel_shader_compile GL_EXT_texture_sRGB_R8 GL_KHR_no_error GL_KHR_texture_compression_astc_sliced_3d GL_MESA_shader_integer_functions GL_ARB_polygon_offset_clamp GL_ARB_texture_filter_anisotropic GL_EXT_semaphore GL_EXT_semaphore_fd GL_KHR_parallel_shader_compile GL_EXT_EGL_image_storage GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_EGL_sync GL_EXT_demote_to_helper_invocation "

[OpenglNoGLSL130]
"Title"="OpenGL shading language version is older than 1.30"

[OpenglNoD3D10]
"Title"="The OpenGL capabilities are insufficient for Direct3D 10 support"

cxdiag64 output:
[Properties]
"vulkan.present.amd64"="yes"

[Properties]
"display.depth"="24"

[Properties]
"display.dpi"="-1.0"

[Properties]
"opengl.vendor"="Intel"

[Properties]
"opengl.version"="2.1 Mesa 21.2.0-devel"

[Properties]
"opengl.renderer"="Mesa Intel(R) HD Graphics (ILK)"

[Properties]
"opengl.glsl_version"="1.20"

[Properties]
"opengl.extensions"="GL_ARB_multisample GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_copy_texture GL_EXT_subtexture GL_EXT_texture_object GL_EXT_vertex_array GL_EXT_compiled_vertex_array GL_EXT_texture GL_EXT_texture3D GL_IBM_rasterpos_clip GL_ARB_point_parameters GL_EXT_draw_range_elements GL_EXT_packed_pixels GL_EXT_point_parameters GL_EXT_rescale_normal GL_EXT_separate_specular_color GL_EXT_texture_edge_clamp GL_SGIS_generate_mipmap GL_SGIS_texture_border_clamp GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_ARB_framebuffer_sRGB GL_ARB_multitexture GL_EXT_framebuffer_sRGB GL_IBM_multimode_draw_arrays GL_IBM_texture_mirrored_repeat GL_3DFX_texture_compression_FXT1 GL_ARB_texture_cube_map GL_ARB_texture_env_add GL_ARB_transpose_matrix GL_EXT_blend_func_separate GL_EXT_fog_coord GL_EXT_multi_draw_arrays GL_EXT_secondary_color GL_EXT_texture_env_add GL_EXT_texture_filter_anisotropic GL_EXT_texture_lod_bias GL_INGR_blend_func_separate GL_NV_blend_square GL_NV_light_max_exponent GL_NV_texgen_reflection GL_NV_texture_env_combine4 GL_S3_s3tc GL_SUN_multi_draw_arrays GL_ARB_texture_border_clamp GL_ARB_texture_compression GL_EXT_framebuffer_object GL_EXT_texture_compression_s3tc GL_EXT_texture_env_combine GL_EXT_texture_env_dot3 GL_MESA_window_pos GL_NV_packed_depth_stencil GL_NV_texture_rectangle GL_ARB_depth_texture GL_ARB_occlusion_query GL_ARB_shadow GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_mirrored_repeat GL_ARB_window_pos GL_ATI_fragment_shader GL_EXT_stencil_two_side GL_EXT_texture_cube_map GL_NV_depth_clamp GL_NV_fog_distance GL_NV_half_float GL_APPLE_packed_pixels GL_ARB_draw_buffers GL_ARB_fragment_program GL_ARB_fragment_shader GL_ARB_shader_objects GL_ARB_vertex_program GL_ARB_vertex_shader GL_ATI_draw_buffers GL_ATI_texture_env_combine3 GL_ATI_texture_float GL_EXT_shadow_funcs GL_EXT_stencil_wrap GL_MESA_pack_invert GL_NV_primitive_restart GL_ARB_depth_clamp GL_ARB_fragment_program_shadow GL_ARB_half_float_pixel GL_ARB_occlusion_query2 GL_ARB_point_sprite GL_ARB_shading_language_100 GL_ARB_sync GL_ARB_texture_non_power_of_two GL_ARB_vertex_buffer_object GL_ATI_blend_equation_separate GL_EXT_blend_equation_separate GL_OES_read_format GL_ARB_color_buffer_float GL_ARB_pixel_buffer_object GL_ARB_texture_compression_rgtc GL_ARB_texture_float GL_ARB_texture_rectangle GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_shared_exponent GL_ARB_framebuffer_object GL_EXT_framebuffer_blit GL_EXT_packed_depth_stencil GL_ARB_vertex_array_object GL_ATI_separate_stencil GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_gpu_program_parameters GL_EXT_texture_array GL_EXT_texture_sRGB_decode GL_EXT_timer_query GL_OES_EGL_image GL_ARB_copy_buffer GL_ARB_depth_buffer_float GL_ARB_draw_instanced GL_ARB_half_float_vertex GL_ARB_instanced_arrays GL_ARB_map_buffer_range GL_ARB_texture_rg GL_ARB_texture_swizzle GL_ARB_vertex_array_bgra GL_EXT_texture_swizzle GL_EXT_vertex_array_bgra GL_NV_conditional_render GL_AMD_seamless_cubemap_per_texture GL_ARB_ES2_compatibility GL_ARB_debug_output GL_ARB_draw_elements_base_vertex GL_ARB_explicit_attrib_location GL_ARB_fragment_coord_conventions GL_ARB_provoking_vertex GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_shader_texture_lod GL_ARB_texture_query_lod GL_ARB_vertex_type_2_10_10_10_rev GL_EXT_provoking_vertex GL_EXT_texture_snorm GL_MESA_texture_signed_rgba GL_NV_copy_image GL_NV_texture_barrier GL_ARB_get_program_binary GL_ARB_robustness GL_ARB_separate_shader_objects GL_EXT_direct_state_access GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_base_instance GL_ARB_compressed_texture_pixel_storage GL_ARB_internalformat_query GL_ARB_map_buffer_alignment GL_ARB_texture_storage GL_AMD_shader_trinary_minmax GL_ARB_clear_buffer_object GL_ARB_copy_image GL_ARB_explicit_uniform_location GL_ARB_invalidate_subdata GL_ARB_program_interface_query GL_ARB_vertex_attrib_binding GL_KHR_debug GL_KHR_texture_compression_astc_ldr GL_ARB_buffer_storage GL_ARB_clear_texture GL_ARB_internalformat_query2 GL_ARB_multi_bind GL_ARB_seamless_cubemap_per_texture GL_ARB_shader_group_vote GL_ARB_shading_language_include GL_ARB_texture_mirror_clamp_to_edge GL_ARB_vertex_type_10f_11f_11f_rev GL_EXT_shader_integer_mix GL_ARB_clip_control GL_ARB_get_texture_sub_image GL_ARB_texture_barrier GL_EXT_polygon_offset_clamp GL_KHR_blend_equation_advanced GL_KHR_context_flush_control GL_ARB_parallel_shader_compile GL_EXT_texture_sRGB_R8 GL_KHR_no_error GL_KHR_texture_compression_astc_sliced_3d GL_ARB_polygon_offset_clamp GL_ARB_texture_filter_anisotropic GL_EXT_semaphore GL_EXT_semaphore_fd GL_KHR_parallel_shader_compile GL_EXT_EGL_image_storage GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_EGL_sync GL_EXT_demote_to_helper_invocation "

[OpenglNoGLSL130.amd64]
"Title"="OpenGL shading language version is older than 1.30"

[OpenglNoD3D10.amd64]
"Title"="The OpenGL capabilities are insufficient for Direct3D 10 support"


***** Fri Jul 16 17:51:22 2021
Starting: '/opt/cxoffice/bin/cxstart' '--bottle' 'nine' '--new-console' '--env' 'D3D_ALWAYS_SOFTWARE=1' '--workdir' 'C:\users\crossover\Start Menu' '--' 'C:\users\crossover\Start Menu\Oblivion.lnk'

CXConfig->read(/opt/cxoffice/etc/cxoffice.conf)
CXConfig->read(/home/io/.cxoffice/cxoffice.conf)
Product version=21.0.0.33696
3177: Grabbing the '/tmp/.wine-1000/bottle-805-142b4b.lock' lock
3177: Got the '/tmp/.wine-1000/bottle-805-142b4b.lock' lock
CXConfig->read(/home/io/.cxoffice/nine/cxbottle.conf)
Mode = 'private'
Bottle environment variables:
 WINEESYNC -> 1
Exec-ing '/opt/cxoffice/bin/cxupdatecheck' '--reminder'
Exec-ing '/opt/cxoffice/bin/cxrating' '--bottle' 'nine' '--reminder'

** Fri Jul 16 17:51:22 2021
Starting '/opt/cxoffice/bin/wineloader' 'winewrapper.exe' '--new-console' '--workdir' 'C:\users\crossover\Start Menu' '--start' '--'
'C:\users\crossover\Start Menu\Oblivion.lnk'

Environment:
  CX_ROOT = "/opt/cxoffice"
  CX_BOTTLE = "nine"
  WINEPREFIX = "/home/io/.cxoffice/nine"
  CX_WINDOWS_VERSION = <undefined>
  PATH = "/opt/cxoffice/bin:/home/io/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
  LD_LIBRARY_PATH = "/opt/cxoffice/lib64:/opt/cxoffice/lib"
  WINEDLLPATH = "/opt/cxoffice/lib/wine"
  WINEDLLOVERRIDES = <undefined>
  LD_PRELOAD = <undefined>
  LD_ASSUME_KERNEL = <undefined>
  WINELOADER = "/opt/cxoffice/bin/wineloader"
  WINESERVER = "/opt/cxoffice/bin/wineserver"
  WINEDEBUG = <undefined>
  CX_LOG = "-"
  CX_DEBUGMSG = <undefined>
  DISPLAY = ":0.0"
  VKD3D_DEBUG = <undefined>
  VKD3D_SHADER_DEBUG = <undefined>
3177: Releasing the '/tmp/.wine-1000/bottle-805-142b4b.lock' lock
Command:
/opt/cxoffice/bin/wineloader winewrapper.exe --new-console --workdir C:\users\crossover\Start Menu --start -- C:\users\crossover\Start Menu\Oblivion.lnk
esync: up and running.
reading '/opt/cxoffice/etc/cxoffice.conf' took 0.0012s
reading '/home/io/.cxoffice/cxoffice.conf' took 0.0003s
3179: Grabbing the '/tmp/.wine-1000/cxoffice.conf.lock' lock
reading '/home/io/.cxoffice/cxoffice.conf' took 0.0000s
'/home/io/.cxoffice/cxoffice.conf' not modified -> no need to save

3179: Releasing the 'cxoffice.conf' lock
reading '/opt/cxoffice/etc/cxoffice.conf' took 0.0008s
reading '/home/io/.cxoffice/cxoffice.conf' took 0.0002s
reading '/home/io/.cxoffice/nine/cxbottle.conf' took 0.0006s
Found no application to rate in nine
0034:fixme:font:opentype_enum_font_names handle name format 1
0034:fixme:font:opentype_enum_font_names handle name format 1
0034:fixme:font:opentype_enum_font_names handle name format 1
0034:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0034:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0068:fixme:font:opentype_enum_font_names handle name format 1
0068:fixme:font:opentype_enum_font_names handle name format 1
0068:fixme:font:opentype_enum_font_names handle name format 1
0070:fixme:font:opentype_enum_font_names handle name format 1
0070:fixme:font:opentype_enum_font_names handle name format 1
0070:fixme:font:opentype_enum_font_names handle name format 1
0068:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0068:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0070:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0070:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0060:fixme:font:opentype_enum_font_names handle name format 1
0060:fixme:font:opentype_enum_font_names handle name format 1
0060:fixme:font:opentype_enum_font_names handle name format 1
0060:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0060:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
002c:fixme:font:opentype_enum_font_names handle name format 1
002c:fixme:font:opentype_enum_font_names handle name format 1
002c:fixme:font:opentype_enum_font_names handle name format 1
002c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0024:fixme:font:opentype_enum_font_names handle name format 1
0024:fixme:font:opentype_enum_font_names handle name format 1
0024:fixme:font:opentype_enum_font_names handle name format 1
0024:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0024:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0024:fixme:exec:SHELL_execute flags ignored: 0x00000100
00fc:fixme:font:opentype_enum_font_names handle name format 1
00fc:fixme:font:opentype_enum_font_names handle name format 1
00fc:fixme:font:opentype_enum_font_names handle name format 1
00fc:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
00fc:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0024:fixme:exec:SHELL_execute flags ignored: 0x00004100
0114:fixme:font:opentype_enum_font_names handle name format 1
0114:fixme:font:opentype_enum_font_names handle name format 1
0114:fixme:font:opentype_enum_font_names handle name format 1
0114:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0114:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0114:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
[1;32mNative Direct3D 9 v0.8.0.385-release is active.
For more information visit https://github.com/iXit/wine-nine-standalone[0m
[1;32mNative Direct3D 9 v0.8.0.385-release is active.
For more information visit https://github.com/iXit/wine-nine-standalone[0m
fixme:d3d9nine:DRIPresentGroup_GetMultiheadCount (0x192cc0), stub!
fixme:d3d9nine:DRIPresentGroup_GetMultiheadCount (0x192cc0), stub!
0114:fixme:rawinput:RegisterRawInputDevices Unhandled flags 0x230 for device 0.
0114:err:winediag:MIDIMAP_drvOpen No software synthesizer midi port found, Midi sound output probably won't work.

**Comment by lorn10** [comment]
I will try an apitrace at the CXBX-R emulator and Baldur's Gate Dark Alliance this weekend. Will see if I can figure this out... ;-)

**Comment by lorn10** [comment]
Okay, it looks that "something" was traced, maybe this really worked. :+1:  Note, this apitrace only contains the **beginning** of the game, so the missing FMV intro sequences and ends at the start screen were the background is missing (not rendered).

My trace can be found here:
https://drive.google.com/file/d/1VlXVGlT4okMNstgIzaV1kiTFmb1sSMpD/view?usp=sharing

Final little note, the CXBX-R "emulator" (effectively it should be called better "Xbox compatibility layer" program) uses currently **DX9** while the original Xbox games are all "**Xbox DX8**" based.

**Comment by axeldavy** [comment]
Thanks for the trace, it was useful to find the bad commit.
Further discussions will be moved to https://gitlab.freedesktop.org/mesa/mesa/-/issues/5201

**Comment by lorn10** [comment]
Last update to that topic, - I am happy to confirm that the mentioned regression is fixed in **Mesa 21.3 devel build c0fc745**. 

Many thanks to @axeldavy for his great effort and the corresponding fix. Finally it turned out that the underlying problem was not in Gallium Nine but at an even "deeper stage" in Gallium itself. 

Whatever, the next time I will be somewhat faster, - I have now learned how an apitrace works. :+1: 
![Screenshot_Mesa_21 3_devel_c0fc745](https://user-images.githubusercontent.com/9931771/129401098-bc8046c5-4e64-40c2-9220-2ab9d4b0158a.png)
